app/streamlit/app.py

```diff
@@ -0,0 +1,125 @@
+import streamlit as st
+import requests
+from streamlit_toggle import st_toggle_switch
+
+st.markdown("<h1 style='text-align: center;'>Chat with the Database</h1>", unsafe_allow_html=True)
+
+st.markdown("""
+    <style>
+    .bubble-user {
+        background-color: #cce7ff;
+        padding: 10px;
+        border-radius: 10px;
+        margin-bottom: 10px;
+        text-align: left;
+        color: black;
+        width: fit-content;
+        margin-left: auto; 
+    }
+    .bubble-database {
+        background-color: #ffccf2;
+        padding: 10px;
+        border-radius: 10px;
+        margin-bottom: 10px;
+        text-align: left;
+        color: black;
+        width: fit-content;
+    }
+    .model-change {
+        text-align: center;
+        margin: 10px 0;
+        font-weight: bold;
+    }
+    </style>
+    """, unsafe_allow_html=True)
+
+if "history" not in st.session_state:
+    st.session_state.history = []
+
+if "input" not in st.session_state:
+    st.session_state.input = ""
+
+if "model" not in st.session_state:
+    st.session_state.model = False
+
+if "use_knowledge_graph" not in st.session_state:
+    st.session_state.use_knowledge_graph = False
+
+for chat in st.session_state.history:
+    if 'model_change' in chat:
+        st.markdown(f"<div class='model-change'>{chat['model_change']}</div>", unsafe_allow_html=True)
+    elif 'knowledge_graph_change' in chat:
+        st.markdown(f"<div class='model-change'>{chat['knowledge_graph_change']}</div>", unsafe_allow_html=True)
+    else:
+        st.markdown(f"<div class='bubble-user'>{chat['user']}</div>", unsafe_allow_html=True) 
+        st.markdown(f"<div class='bubble-database'>{chat['response']}</div>", unsafe_allow_html=True)
+
+def send_message():
+    user_input = st.session_state.input
+    if user_input:
+        model_name = "openai" if st.session_state.model else "mistral"
+        endpoint = "http://127.0.0.1:8000/generate_sparql/" if st.session_state.use_knowledge_graph else "http://127.0.0.1:8000/generate_sql/"
```

Comment:
@BelenA25 please modify the base url to use envdot please to manage differents urls